HOWTO for release process

dmoody256 opened the pull request:
Some of the test will fail on windows in the appveyor build vm's. This PR is for modify the failing tests so they should throw no result or correctly execute the test in the appveyor environment.

This branches from PR: https://github.com/SCons/scons/pull/33.

Below are list of tests which were failing in the appveyor [build for PR 33](https://ci.appveyor.com/project/dmoody256/scons/build/1.0.102) and now are passing for the latest commit:
https://ci.appveyor.com/project/bdbaddog/scons/build/1.0.5

They have been updated along with the commits related as shown below:

**Clang tests:**
- ff135a9
- dc997f8
- ffda18f

**Pkg-config test:**
- ffda18f

**M4 test:**
- 1381c69

**msvs 10 exec test:**
- 893c0d8

**JobTests workaround**
- d6fce4c
- 98c53e5
- 9dde17b

**MSVC UWP test**
- 2698634
- d176632

**SWIG test**
- d176632

**Docbook py3 test**
- b1d21ec

**Interactive py3 tests**
- ac50932

**YACC and gettext tests**
- fa0583a

bdbaddog commented on test/Clang/clang_default_environment.py:
```diff
@@ -25,13 +25,18 @@
 if not test.where_is('clang'):
     test.skip_test("Could not find 'clang', skipping test.\n")
 
+platform = Base()['PLATFORM']
```
Doesn't the above where_is('clang') take care of this?

dmoody256 replied:
default environment on windows will have the vcvarsall.bat run which will not taken into account the original environment and path that test.where_is sees

bdbaddog commented on test/Clang/clang_shared_library.py:
```diff
@@ -41,15 +43,18 @@
     libraryname = 'foo.dll'
+    # add the environment, otherwise the environment will consist of only vcvarsall.bat variables
+    # and not clang
+    env_str = "import os\nenv = Environment(tools=['clang', 'link'], ENV = os.environ)"
```
Why do this only for win32?

dmoody256 replied:
The Environment on windows was created without the original environment that the runtest.py was called from. Not sure if that is an appveyor specific issue or something weird with the msvs tool running vcvarsall.bat, but the mingw tools for linking were missing from the path causing a build failure on the link step during the test.

Adding the os.environ here retained the original environment and path that runtest.py was run from.

bdbaddog commented on test/Clang/clang_specific_environment.py:
```diff
@@ -25,17 +25,25 @@
     test.skip_test("Could not find 'clang', skipping test.\n")
 
+env_str = "env = Environment(tools=['clang', 'link'])"
+platform = Base()['PLATFORM']
```
better to use sys.platform == for this rather than adding additional dependency on Base.. which can change or be removed.

dmoody256 replied:
agreed, I used the scons way for consistency with other tests, but I would prefer sys.platform.

bdbaddog commented:
Your changes will disable and/or break tests which do complete successfully on our windows buildbot workers and on properly configured machines.

See: http://buildbot.scons.org/#/builders/8/builds/19

Plus I really don't like huge pull requests which affect many different tests.  Ideally you'd address the comments and make sure that the changes aren't disabling tests which work outside the appveyor environment.

The goal is not that only the tests which function in the appveyor environment run, but rather that tests which can pass will run in any environment which can run them and get skipped in those that cannot.

dmoody256 commented:
> Your changes will disable and/or break tests which do complete successfully on our windows buildbot workers and on properly configured machines.
>
>See: http://buildbot.scons.org/#/builders/8/builds/19

None of the failed test in the buildbot you linked were modified in this PR, and I dont see any underlying files in this PR which could have affected those tests on buildbot.

> Plus I really don't like huge pull requests which affect many different tests. Ideally you'd address the comments and make sure that the changes aren't disabling tests which work outside the appveyor environment.
>
> The goal is not that only the tests which function in the appveyor environment run, but rather that tests which can pass will run in any environment which can run them and get skipped in those that cannot.

I addressed your review comments, if there are certain changes in this PR you feel are changing the tests to disable or break them in other environments, we could check one of the appveyor environment variables and skip the test in those cases for appveyor so as to leave the test the same on other environments, but still be able to get passing results from appveyor.
